packages/contracts/contracts/voting/addresslist/Addresslist.sol

```diff
@@ -23,24 +23,14 @@ abstract contract Addresslist {
     /// @param member The array of member addresses to be added or removed.
     error InvalidAddresslistUpdate(address member);
 
```

Comment:
Yes, it is technically possible.
I thought about it as well but decided against it. In some cases, `Addresslist` could be used in a registry contract so that it should emit the `event MembershipContractAnnounced` and implement `function isMember`.
That depends on the specific use case, and I wouldn't make it mandatory here.

The `abstract contract Addresslist` should, IMO, not emit events.

Reply:
why would registry need to inherit from address list ? what kind of registry are we talking about ?  even if so, `MembershipContractAnnounced` wouldn't be emitted from addAddresses and removeAddresses functions, so why not still emitting MembersAdded and MembersRemoved emitted from `addresslist` ?

Reply:
`Addresslist` is a contract to be reused in different context and we don't know it yet. Maybe someone wants no event to be emitted. It is wrong to put all the events in here already.
Events should to happen on the concrete implementation, not the abstract one.

Reply:
No, 

Look at OZ's Governance Contract. Then, you should tell them not to emit events from there as billions of people use it and they might not need to emit anything.

Reply:
OZ governor is a different thing. It is basically a DAO that you can extend with a few modules.
Show me the case where they inherit events from two, one, or none out of two interfaces.

Reply:
Done as part of 541fc3a08797768dafcba7ac8372b19e48c1af71